frontend/src/metabase/entities/bookmarks.js

```diff
@@ -51,12 +51,12 @@ const Bookmarks = createEntity({
         type,
         item_id,
       }));
+      dispatch({ type: REORDER_ACTION, payload: bookmarks });
```

Comment:
One thing about dispatching this event optimistically is that we don't handle the scenario where the request fails for some reason. If we want to be optimistic about this update, we should probably catch any incoming errors and revert things back to their previous state.

Reply:
Added this functionality, with an e2e test